manifests/pulp.pp

```diff
@@ -49,14 +62,28 @@
     repo_auth              => true,
     puppet_wsgi_processes  => 1,
     enable_katello         => true,
+    https_cert             => $https_cert,
+    https_key              => $https_key,
+    ca_cert                => $ca_cert,
     subscribe              => Class['certs', 'certs::qpid_client'],
   }
 
   contain ::pulp
 
-  foreman::config::passenger::fragment { 'pulp':
-    content     => file('katello/pulp-apache.conf'),
-    ssl_content => file('katello/pulp-apache-ssl.conf'),
+  if $manage_httpd {
+    pulp::apache::fragment { 'httpd_pub':
+      ssl_content => file('katello/pub-apache.conf'),
+    }
+  } else {
+    foreman::config::passenger::fragment { 'pulp':
+      content     => file('katello/pulp-apache.conf'),
+      ssl_content => file('katello/pulp-apache-ssl.conf'),
+    }
+
+    foreman::config::passenger::fragment { 'pub':
+      content     => file('katello/pub-apache.conf'),
+      ssl_content => file('katello/pub-apache.conf'),
+    }
```

Comment:
Do we also need to proxy GPG to the foreman host?

Reply:
I don't think so because candlepin runs on the host itself so there's no need to proxy it. If they are split, we might need to figure out what's correct. If a smart proxy host proxies it to an application host which proxies it to a candlepin host that's somewhat complexer than it might need to be.

Reply:
This will cause problems though -- https://github.com/theforeman/puppet-pulp/commit/5ff7611c6e577b95f5a3f096830c4e3097c4db2d